### PR TITLE
Resolve Promises using allSettled method

### PR DESCRIPTION
**Pull request by algtan**
Refactor `swapMinAmounts` to resolve Promises using `Promise.allSettled` instead of a for loop that awaits each swap quote fetch.

### CODE REVIEW

**Comment by LPeterson18** [comment]
Everything looks good to me.

File: src/utils/swapMinAmounts.ts [hunk 1]
```diff
@@ -59,42 +77,73 @@ export async function swapMinAmounts(
...
+    swapQuoteParamArr.push({
+      fromCurrencyWallet,
+      toCurrencyWallet,
+      plugin
+    })
+  }
+
+  const minAmountPromiseArr = swapQuoteParamArr.map(async swapQuoteParam => {
+    const { fromCurrencyWallet, toCurrencyWallet, plugin } = swapQuoteParam
+    const disabledPluginMap = createDisabledSwapPluginMap(
```
Comment (algtan) [inline src/utils/swapMinAmounts.ts:92]: No need to recreate for each plugin. Instead, create one global map.

File: src/utils/swapMinAmounts.ts [hunk 2]
```diff
@@ -59,42 +77,73 @@ export async function swapMinAmounts(
...
-            toWallet: toCurrencyWallet,
-            amount: nativeAmount
-          })
-          await account.fetchSwapQuote(binarySearchSwapReq, requestOptions)
-          return true
-        } catch (e) {
-          if (e.message !== 'Amount is too low') throw e
-          return false
-        }
+    swapQuoteParamArr.push({
```
Comment (algtan) [inline src/utils/swapMinAmounts.ts:83]: This loop is unnecessary

File: src/utils/swapMinAmounts.ts [hunk 3]
```diff
@@ -59,42 +77,73 @@ export async function swapMinAmounts(
...
+      initBinarySearchMin,
+      initBinarySearchMax
+    )
+  })
+
+  const settledMinAmounts = await Promise.allSettled(minAmountPromiseArr)
+
+  const minAmountArr: MinAmountInfo[] = []
+
+  for (let i = 0; i < settledMinAmounts.length; i++) {
```
Comment (algtan) [inline src/utils/swapMinAmounts.ts:125]: Use filter

File: src/utils/swapMinAmounts.ts (hunk 1 above)
Comment (algtan) [reply src/utils/swapMinAmounts.ts:92]: https://github.com/EdgeApp/edge-core-js/blob/c9747591ca36ae95e193ff50c157a45d18123c69/src/types/types.js#L882

**Comment by algtan** [comment]
This PR is superseded by https://github.com/EdgeApp/edge-swap-server/pull/24.

**Pull request closed without merging**